### PR TITLE
4886 correct link check ignore doc

### LINKED ISSUE #4886: [docs][question] what is valid for linkcheck_anchors_ignore?
Subject: unable to wield `linkcheck_anchors_ignore`, likely user error.

### Problem

From the [description of `linkcheck_anchors_ignore`](http://www.sphinx-doc.org/en/master/config.html#confval-linkcheck_anchors_ignore):

> A list of regular expressions that match URIs that should skip checking the validity of anchors in links. This allows skipping entire sites, where anchors are used to control dynamic pages, or just specific anchors within a page, where javascript is used to add anchors dynamically, or use the fragment as part of to trigger an internal REST request. Default is `["/#!"]`.

An example of this would be linking to an anchor on a GitHub README.  One of the links I am working with: [https://github.com/jonmiles/bootstrap-treeview#onhovercolor](https://github.com/jonmiles/bootstrap-treeview#onhovercolor).

So I thought maybe something like

```py
linkcheck_anchors_ignore = [
    r'https://github.com/jonmiles/bootstrap-treeview'
]
```

would work, or `.*bootstrap-treeview.*`, or many others.  But I cannot seem to get a regular expression that will work, and instead need to enumerate all of the anchors in `linkcheck_ignore` explicitly.

#### Procedure to reproduce the problem

With an `index.rst` and fresh sphinx project

```rst
Welcome
=======

See `this page <https://github.com/jonmiles/bootstrap-treeview#onhovercolor>`_
for more information.
```

`make linkcheck` will (correctly) report an invalid anchor.  But no `linkcheck_anchors_ignore` values I have tried in `conf.py` work.

#### Expected results

I feel like `[r'https://github.com/jonmiles/bootstrap-treeview#*']` should be accepted here.

### PR DESCRIPTION
**Pull request by TimKam**
Subject: Correct link check ignore doc

### Feature or Bugfix
- Docs/tests

### Purpose
- Adds tests to better cover current behavior of link check ignore config
- Corrects link check ignore documentation

### Relates
- Fixes #4886

### CODE REVIEW

File: doc/usage/configuration.rst [hunk 1]
```diff
@@ -2485,12 +2485,20 @@ Options for the linkcheck builder
...
+   internal REST request. Default is ``["^!"]``.
+
+   .. note::
+
+      If you want to ignore anchors of a specific page or of pages that match a
+      specific pattern, use :confval:`linkcheck_ignore` instead, for example as
+      follows::
+
+         linkcheck_ignore = [
+            'http://www.sphinx-doc.org/en/1.7/intro.html#*'
```
Comment (tk0miya) [inline doc/usage/configuration.rst]: I think `#*` is not needed here. Because linkcheck builder uses `re.match()` to check URLs.

**Review by tk0miya — APPROVED** [review]
LGTM with nits

File: doc/usage/configuration.rst [hunk 2]
```diff
@@ -2485,12 +2485,20 @@ Options for the linkcheck builder
...
+   internal REST request. Default is ``["^!"]``.
+
+   .. note::
+
+      If you want to ignore anchors of a specific page or of pages that match a
+      specific pattern, use :confval:`linkcheck_ignore` instead, for example as
+      follows::
+
+         linkcheck_ignore = [
+            'http://www.sphinx-doc.org/en/1.7/intro.html#'
```
Comment (tk0miya) [inline doc/usage/configuration.rst]: This means hyperlink to `intro.html` is link-checked, but `intro.html#foobar` is ignored. Is this intended?

Comment (TimKam) [reply doc/usage/configuration.rst]: Exactly, this is intended. The example shows how users can ignore all anchors on a specific page, without ignoring the page itself.

Comment (TimKam) [reply doc/usage/configuration.rst]: I clarified the doc accordingly.

**Review by tk0miya — APPROVED** [review]
LGTM!

**Pull request merged**